### DISCUSSION TITLE
How to use nested layout in nuxt3?

### DISCUSSION THREAD

**Discussion by FatNerdPeng** [nuxt/nuxt#19476] [Questions]
Hello,my attempt to use another layout in a layout failed like this:

![image](https://user-images.githubusercontent.com/33616855/223069026-92789b71-6024-48c2-a3bb-0ceb4cca039c.png)
![image](https://user-images.githubusercontent.com/33616855/223069101-f9dd01cd-97e5-40ee-aece-7cdb502e29d4.png)
![image](https://user-images.githubusercontent.com/33616855/223069163-1cca8cf7-27fa-485b-90a4-7b32643dab74.png)
![image](https://user-images.githubusercontent.com/33616855/223069234-e5d4b8a6-9d05-453d-a005-ac5e93fb6a89.png)
![image](https://user-images.githubusercontent.com/33616855/223069301-4d5faf22-b297-4859-b144-f8be63786098.png)
![image](https://user-images.githubusercontent.com/33616855/223069362-18146704-9d1d-4a33-9051-08db516f9db8.png)
![image](https://user-images.githubusercontent.com/33616855/223069423-ec3a56fb-2f09-4dbc-a87f-443a07814de8.png)
![image](https://user-images.githubusercontent.com/33616855/223069466-eadaa36c-f822-46ac-8b11-e5478e3bc602.png)

minimal reproduction: https://stackblitz.com/edit/github-h52b5t

As shown in the figure, the index.vue and other.vue under the body directory is missing the aside and top. I don't know where the problem is.
The reason why I want to do this is because the demand is like this:
1.Each page under pages directory should have main.vue layout, but only specific pages have components <SomeComp>
2.For pages with components <SomeComp>, only the page content is switched, not the components data.
I know that I can achieve these requirements through components and vuex, but I think that's too complicated. I think it's more convenient to use nested layout,so I tried this and didn't know what I did wrong.
Thanks.

**Comment by imlautaro** [comment]
I think the way you can use a layout inside another one is by using the `<NuxtLayout />` component instead of the `definePageMeta`.

For example, in your body layout, remove the `definePageMeta` and wrap your layout with `<NuxtLayout name="main">...</NuxtLayout>`

```vue
<!-- ~/layouts/body.vue -->
<template>
  <div>
    <NuxtLayout name="main">
      <span>Body Layout Here!</span>
      <slot />
    </NuxtLayout>
  </div>
</template>
```

**Reply by FatNerdPeng** [reply]
I tried it again as you said. It's really OK,thank you!
